### DISCUSSION TITLE
Elastoplastic model for solid analaysis.

### DISCUSSION THREAD

**Discussion by HoseongJeong** [kinnala/scikit-fem#1157] [Q&A] [answered]
I am conducting solid FEA research using your code, and thanks for your codes!!  (It is beneficial for my research!)

I want to try cyclic load analysis of solid material.

Thus I need to apply the elastoplastic material model (it depends on load history) which can save the strain history of each element.

Is it possible to implement the elastoplastic material model in Bilinear form? (for example, using a class object, etc...)

**Comment by bhaveshshrimali** [comment, marked as answer]
Elasto-plasticity should be straightforward to implement. I haven't personally done it, but I did implement rate-dependent large-strain viscoelasticity back in the day. https://github.com/kinnala/scikit-fem/discussions/730#discussioncomment-1267910

Conceptually the two are similar in that you have an internal variable (plastic/viscoelastic strain) whose evolution is described by ODE in time (e.g. flow rule in plasticity).

All you need to do is discretize your flow rule, implement the expression for tangent and return mapping for stress, and you should be able to assemble it into bilinear form (tangent) and residual with your own newton solver (if required).

If you have a specific material model in mind, I could help construct an example

**Reply by HoseongJeong** [reply]
Thanks a lot!!! It is really helpful to me!!!!!
